src/main/kotlin/app/batch/TransformationProcessor.kt

```diff
@@ -0,0 +1,49 @@
+package app.batch
+
+import app.domain.Record
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonObject
+import org.springframework.batch.item.ItemProcessor
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+
+/**
+ * Perform any collection specific transformations prior to output.
+ *
+ * Currently, only equality data needs this
+ */
+@Component
+class TransformationProcessor: ItemProcessor<Record, Record> {
+
+    /**
+     * Perform any collection specific transformations prior to output.
+     */
+    override fun process(item: Record): Record? =
+        when (topicName) {
+            EQUALITY_TOPIC -> {
+                transformedEqualityRecord(item)
+            }
+            else -> {
+                item
+            }
+        }
+
+    private fun transformedEqualityRecord(item: Record): Record =
+        Record(gson.toJson(wrapped(item)), item.manifestRecord)
+
+    private fun wrapped(item: Record): JsonObject =
+        JsonObject().apply { add("message", wrappedObject(item)) }
+
+    private fun wrappedObject(item: Record): JsonObject =
+        gson.fromJson(item.dbObjectAsString, JsonObject::class.java).apply {
+            addProperty("@type", item.manifestRecord.externalInnerSource)
```

Comment:
It can't be null, it is declared in the data class as a String, rather than a nullable String (i.e. `String` not `String?` ) so any attempt to initialise with `null` would result in a `NullPointerException` upstream. So in short at this point we are safe.